code/go/0chain.net/smartcontract/zcnsc/authorizer.go

```diff
@@ -53,24 +54,15 @@ func (zcn *ZCNSmartContract) AddAuthorizer(
 		Logger.Error("public key error", zap.Error(err))
 		return "", err
 	}
-
-	if params.ID == "" {
-		err = common.NewError(code, "authorizer ID cannot be empty")
-		Logger.Error("authorizer id error", zap.Error(err))
-		return "", err
-	}
-
-	authorizerID = params.ID
-
 	if params.PublicKey == "" {
 		err = common.NewError(code, "public key was not included with transaction")
 		Logger.Error("public key error", zap.Error(err))
 		return "", err
 	}
 
 	publicKeyBytes, _ := hex.DecodeString(params.PublicKey)
```

Comment:
We'd better not ignore the error for hex decoding, otherwise wrong authorizerID would be added.

Reply:
fixed in different PR